myst_nbs/gaussian_processes/gaussian_process.myst.md

```diff
@@ -96,84 +90,97 @@ N = len(y)
-sns.heatmap(Σ.tag.test_value, xticklabels=False, yticklabels=False)
+sns.heatmap(sigma.eval(), xticklabels=False, yticklabels=False);
 ```
 
-The following generates predictions from the GP model in a grid of values:
+The following generates predictions from the Gaussian Process model in a grid of values:
 
 ```{code-cell} ipython3
-with gp_fit:
+# Prediction over grid
+xgrid = np.linspace(-6, 6)
+D_pred = squared_distance(xgrid, xgrid)
+D_off_diag = squared_distance(x, xgrid)
 
-    # Prediction over grid
-    xgrid = np.linspace(-6, 6)
-    D_pred = squared_distance(xgrid, xgrid)
-    D_off_diag = squared_distance(x, xgrid)
+gp_fit.add_coords({"pred_id": xgrid, "pred_id2": xgrid})
 
+with gp_fit as gp:
     # Covariance matrices for prediction
-    Σ_pred = η_sq * T.exp(-ρ_sq * D_pred)
-    Σ_off_diag = η_sq * T.exp(-ρ_sq * D_off_diag)
+    sigma_pred = eta_sq * at.exp(-rho_sq * D_pred)
+    sigma_off_diag = eta_sq * at.exp(-rho_sq * D_off_diag)
 
     # Posterior mean
-    μ_post = Deterministic("μ_post", T.dot(T.dot(Σ_off_diag, matrix_inverse(Σ)), y))
+    mu_post = pm.Deterministic(
+        "mu_post", at.dot(at.dot(sigma_off_diag, pm.math.matrix_inverse(sigma)), y), dims="pred_id"
+    )
     # Posterior covariance
-    Σ_post = Deterministic(
-        "Σ_post", Σ_pred - T.dot(T.dot(Σ_off_diag, matrix_inverse(Σ)), Σ_off_diag.T)
+    sigma_post = pm.Deterministic(
+        "sigma_post",
+        sigma_pred
+        - at.dot(at.dot(sigma_off_diag, pm.math.matrix_inverse(sigma)), sigma_off_diag.T),
+        dims=("pred_id", "pred_id2"),
     )
 ```
 
 ```{code-cell} ipython3
 with gp_fit:
-    svgd_approx = pm.fit(400, method="svgd", inf_kwargs=dict(n_particles=500))
+    svgd_approx = pm.fit(400, method="svgd", inf_kwargs=dict(n_particles=100))
 ```
 
 ```{code-cell} ipython3
 gp_trace = svgd_approx.sample(1000)
 ```
 
 ```{code-cell} ipython3
-traceplot(gp_trace, var_names=["η_sq", "ρ_sq", "σ_sq"]);
+az.plot_trace(gp_trace, var_names=["eta_sq", "rho_sq", "sigma_sq"]);
 ```
 
-Sample from the posterior GP
+Sample from the posterior Gaussian Process
 
 ```{code-cell} ipython3
-y_pred = [
-    np.random.multivariate_normal(m, S) for m, S in zip(gp_trace["μ_post"], gp_trace["Σ_post"])
-]
+post = az.extract(gp_trace, num_samples=200)
+
+y_pred = multivariate_normal(
+    post["mu_post"], post["sigma_post"], dims=("pred_id", "pred_id2")
+).rvs()
 ```
 
 ```{code-cell} ipython3
-for yp in y_pred:
-    plt.plot(np.linspace(-6, 6), yp, "c-", alpha=0.1)
-plt.plot(x, y, "r.")
+_, ax = plt.subplots(figsize=(12, 8))
+ax.plot(xgrid, y_pred.transpose(..., "sample"), "c-", alpha=0.1)
+ax.plot(x, y, "r.");
 ```
 
+## Authors
+*  Adapted from Stan's [example-models repository](https://github.com/stan-dev/example-models/blob/master/misc/gaussian-process/gp-fit.stan) by Chris Fonnesbeck in 2016
+* Updated by Ana Rita Santos and Sandra Meneses in July, 2022 ([pymc#404](https://github.com/pymc-devs/pymc/pull/404))
+
 ```{code-cell} ipython3
```

Comment:
There should be a watermark title right above this cell: https://docs.pymc.io/en/latest/contributing/jupyter_style.html#watermark